site/static/examples/kind-with-registry.sh

```diff
@@ -36,7 +36,7 @@ EOF
 # We want a consistent name that works from both ends, so we tell containerd to
 # alias localhost:${reg_port} to the registry container when pulling images
 REGISTRY_DIR="/etc/containerd/certs.d/localhost:${reg_port}"
-for node in $(kind get nodes); do
+for node in $(kubectl get nodes -o jsonpath='{.items[*].metadata.name}'); do
```

Comment:
isn't the right fix to make `kind get nodes` work with podman?

Reply:
podman works with kind get nodes, it logs the experimental disclaimer ... podman is not stable enough and still lacks some important capabilities https://github.com/kubernetes-sigs/kind/issues/1778

Reply:
There's some small discussion on the linked issue that highlights why the current `kind get nodes` output is a challenge with podman.

Reply:
oh, my bad again, it does rely on `kubectl` at the end of the script ... nevermind, let me reassing this to @BenTheElder , he knows better this script

/assign @BenTheElder

Reply:
> podman works with kind get nodes, it logs the experimental disclaimer ... podman is not stable enough and still lacks some important capabilities https://github.com/kubernetes-sigs/kind/issues/1778

the log output should not be capture here, are we logging to stdout??

Reply:
it uses the logger, I was trying to remember what we log to stderr and what to stdout, I remember some issues with that before

Reply:
It should be stderr, which should not be captured:

https://github.com/kubernetes-sigs/kind/blob/0483056594cb49bec92dc5c3f396e7390a39765f/pkg/internal/runtime/runtime.go#L16

https://github.com/kubernetes-sigs/kind/blob/0483056594cb49bec92dc5c3f396e7390a39765f/pkg/cmd/logger.go#L31-L36

Reply:
https://github.com/kubernetes-sigs/kind/blob/0483056594cb49bec92dc5c3f396e7390a39765f/pkg/cluster/internal/providers/podman/provider.go#L45

Reply:
```
$ (out="$(KIND_EXPERIMENTAL_PROVIDER=podman kind get nodes || true)"; echo "out: $out")
using podman due to KIND_EXPERIMENTAL_PROVIDER
enabling experimental podman provider
ERROR: failed to list nodes: command "podman ps -a --filter label=io.x-k8s.kind.cluster=kind --format '{{.Names}}'" failed with error: exec: "podman": executable file not found in $PATH
Command Output: 
out: 
```

Are we sure the script is broken??

Reply:
> Are we sure the script is broken?

For podman running on macOS, yes, but I may have misunderstood the reason. Here's what I'm seeing when the script starts to `docker exec` the nodes:

```
using podman due to KIND_EXPERIMENTAL_PROVIDER
enabling experimental podman provider
Error: can only create exec sessions on running containers: container state improper
```

@BenTheElder you're right that the two lines about podman are not going to stdout. It must be a race condition, and since my solution involves an API call, it's slow enough to lose the race.

Reply:
Interesting, we really should be safe to exec once `kind create cluster` succeeds ... uhhh

Did the container crash or something?

Let's follow up more on the issue.